activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -127,7 +127,12 @@ def rollback_records
       end
 
       def before_commit_records
-        records.uniq.each(&:before_committed!) if @run_commit_callbacks
+        return unless @run_commit_callbacks
+
+        deterministic_order = records.uniq.sort_by do |r|
+          r.class.name + r.id.to_s rescue "_"
```

Comment:
What is being rescued here?

Reply:
Probably unnecessary: this was just extra safety, so it doesn't affect existing commits by failing to sort.

Reply:
I perfer to not have blanket rescues in the code. What you call safety now will be actually a source of really hard to track bugs.

Reply:
Thanks for the review! Agreed and removed. (squashed)